### PR TITLE
add waf tests

### PR DESCRIPTION
**Pull request by gabelton**
https://uktrade.atlassian.net/browse/DBTP-301

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -57,3 +57,11 @@ repos:
...
         pass_filenames: false
         always_run: true
+  - repo: local
+    hooks:
+      - id: check-tracked-files
+        name: Check tracked file changes
+        entry: git diff --name-status --exit-code
+        language: system
+        pass_filenames: false
+        always_run: true
```
Comment (yusufsheiqh) [inline .pre-commit-config.yaml:67]: All of this can be removed. You may want to rebase `main`.

File: tests/test_waf_cli.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+            "--svc",
+            "svc",
+            "--env",
+            "env",
+            "--waf-path",
+            "waf.yml",
+        ],
+    )
+
+    assert "CloudFormation Stack already exists" in result.output
```
Comment (yusufsheiqh) [inline tests/test_waf_cli.py:152]: Should we assert on the exit code as well?

**Review by yusufsheiqh — COMMENTED** [review]
Great work on getting these unit tests in place!

File: commands/waf_cli.py [hunk 3]
```diff
@@ -68,6 +70,20 @@ def attach_waf(app, project_profile, svc, env):
...
+    return cf_client.create_stack(
+        StackName=f"{app}-{svc}-{env}-CustomWAFStack",
+        TemplateBody=raw,
+        TimeoutInMinutes=5,
+        OnFailure="DELETE",
+        Tags=[
+            {"Key": "Name", "Value": "copilot tools custom waf"},
+        ],
+        ClientRequestToken=uuid.uuid4().hex,
+        EnableTerminationProtection=False,
```
[the remark this answers is not in the document]
Comment (gabelton) [reply commands/waf_cli.py:83]: Maybe a question for @ejayesh , as I've just extracted his code into a more testable helper method

[the remark this answers is not in the document]
Comment (ejayesh) [reply commands/waf_cli.py:83]: For now we are not sure, we want this to be false so its easier to remove the CF template.  We can change this to True later.

**Pull request merged**